### PR TITLE
#170947637 Get trip statistics

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
display statistics of the trips have been made
#### Description of Task to be completed?
- show a user number of trips he/she has made based on the date of the first travel he/she made up to the current date
- show him/her the last days since from the first day he/she started to trave up to now
- show a user the details of the trips he/she made
- show a user total number of trips he/she made
- show a user total number of request he/she created
- show a user total number of trip requests were rejected and pending
#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git fetch` command to download commits, files, and refs from a remote repository into your local repo.
- run `git ft-trip-made-in-x-time-flame-170947637` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run dev-start` to start the application.
- do login with this default user
```
 "email":"knights@gmail.com",
  "password":"Niyonkuru@1"
```
- browse to `http://localhost:8080/stats` to access your trip statistics.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947637](https://www.pivotaltracker.com/story/show/170947637)
#### Screenshots (if appropriate)
Screen Shot 2020-06-04 at 16 42 16

### CODE REVIEW

**Comment by T2Wil** [comment]
Nice work @EugeneMunya , I would suggest that you may add space between the navbar and the page body.

**Comment by EugeneMunya** [comment]
> Nice work @EugeneMunya , I would suggest that you may add space between the navbar and the page body.

Let me work on it

**Comment by EugeneMunya** [comment]
> Nice work @EugeneMunya , I would suggest that you may add space between the navbar and the page body.

I have done to implement the feedback check again

**Review by T2Wil — APPROVED** [review]
Nice work Eugene.

**Review by Kagorora — APPROVED** [review]

File: src/components/getStats/getStats.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  async UNSAFE_componentWillMount() {
+    await this.props.callApiThunk('get', '/trips/myrequest', allMyrequests);
+    this.setState({
+      allRequests: this.props.userStats.allMyRequest,
+      data: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved'),
+      pending: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'pending'),
+      rejected: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'rejected'),
+      trips: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved').length,
+      startDate: (this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved').length) ? this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved')[0].departureDate : 0,
```
Comment (ceelogre) [inline src/components/getStats/getStats.js]: According to your ternary operator, what does having a startDate of zero mean?

Comment (EugeneMunya) [reply src/components/getStats/getStats.js]: That is the first date of the first trip you traveled, if you haven't done any trip, startDate will be 0

**Review by Niyonsengaeric — APPROVED** [review]

File: src/components/getStats/getStats.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import { connect } from 'react-redux';
+import moment from 'moment';
+import { Table, Badge } from 'react-bootstrap';
+import { callApiThunk } from '../../redux/thunk/index';
+import { allMyrequests } from '../../redux/actions/requestActions';
+import Nav from '../Nav';
+import { Stats, TableHeader } from '../../util/reusableTableHeader';
+
+class numberOfTrips extends Component {
+  constructor(props) {
```
Comment (kagabof) [inline src/components/getStats/getStats.js]: Please user es6

File: src/components/getStats/getStats.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    this.setState({
+      allRequests: this.props.userStats.allMyRequest,
+      data: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved'),
+      trips: this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved').length,
+      startDate: (this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved').length) ? this.props.userStats.allMyRequest.filter((trip) => trip.status === 'approved')[0].departureDate : 0,
+    });
+  }
+
+
+  render() {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/getStats/getStats.js:43]: please solve this.

**Review by kagabof — CHANGES_REQUESTED** [review]

**Pull request closed without merging**